src/puya/awst_build/module.py

```diff
@@ -576,6 +579,34 @@ def _process_contract_class_options(
                             )
                         else:
                             scratch_slot_reservations |= slots
+                case "state_totals":
+                    match kw_expr:
+                        case mypy.nodes.StrExpr(value="auto"):
+                            pass
```

Comment:
hmm, I think we still want to warn if the user doesn't specify `"auto"` explicitly, and they're inheriting from a class that does have a `state_totals` - which means this would need to set some marker here rather than just passing I guess?

Reply:
Have replaced `any_defined` with an `is_explicit` flag that is set to False when no value is provided